Description: patches to make virtualbox work with new mesa GL API
--- virtualbox-5.2.8-dfsg.orig/src/VBox/GuestHost/OpenGL/glapi_parser/APIspec.txt
+++ virtualbox-5.2.8-dfsg/src/VBox/GuestHost/OpenGL/glapi_parser/APIspec.txt
@@ -10249,6 +10249,15 @@ param       framebuffer     GLuint
 category    GL_EXT_framebuffer_object
 chromium    extpack
 
+name        BindFramebuffer
+alias       BindFramebufferEXT
+return      void
+param       target          GLenum
+paramprop   target          GL_FRAMEBUFFER_EXT GL_READ_FRAMEBUFFER_EXT GL_DRAW_FRAMEBUFFER_EXT
+param       framebuffer     GLuint
+category    GL_EXT_framebuffer_object
+chromium    extpack
+
 name        DeleteFramebuffersEXT
 return      void
 param       n               GLsizei
@@ -10264,6 +10273,15 @@ category    GL_EXT_framebuffer_object
 props       get
 chromium    extpack
 
+name        GenFramebuffers
+alias       GenFramebuffersEXT
+return      void
+param       n               GLsizei
+param       framebuffers    GLuint *
+category    GL_EXT_framebuffer_object
+props       get
+chromium    extpack
+
 name        CheckFramebufferStatusEXT
 return      GLenum
 param       target          GLenum
@@ -10298,6 +10316,20 @@ return      void
 category    GL_EXT_framebuffer_object
 chromium    extpack
 
+name        FramebufferTexture2D
+alias       FramebufferTexture2DEXT
+return      void
+param       target          GLenum
+paramprop   target          GL_FRAMEBUFFER_EXT GL_READ_FRAMEBUFFER_EXT GL_DRAW_FRAMEBUFFER_EXT
+param       attachment      GLenum
+paramprop   attachment      GL_COLOR_ATTACHMENT0_EXT GL_COLOR_ATTACHMENT1_EXT GL_COLOR_ATTACHMENT2_EXT GL_COLOR_ATTACHMENT3_EXT GL_COLOR_ATTACHMENT4_EXT GL_COLOR_ATTACHMENT5_EXT GL_COLOR_ATTACHMENT6_EXT GL_COLOR_ATTACHMENT7_EXT GL_COLOR_ATTACHMENT8_EXT GL_COLOR_ATTACHMENT9_EXT GL_COLOR_ATTACHMENT10_EXT GL_COLOR_ATTACHMENT11_EXT GL_COLOR_ATTACHMENT12_EXT GL_COLOR_ATTACHMENT13_EXT GL_COLOR_ATTACHMENT14_EXT GL_COLOR_ATTACHMENT15_EXT GL_DEPTH_ATTACHMENT_EXT GL_STENCIL_ATTACHMENT_EXT
+param       textarget       GLenum
+param       texture         GLuint
+param       level           GLint
+return      void
+category    GL_EXT_framebuffer_object
+chromium    extpack
+
 name        FramebufferTexture3DEXT
 return      void
 param       target          GLenum
@@ -10427,6 +10459,24 @@ name        BlitFramebufferEXT
 return      void
 param       srcX0       GLint
 param       srcY0       GLint
+param       srcX1       GLint
+param       srcY1       GLint
+param       dstX0       GLint
+param       dstY0       GLint
+param       dstX1       GLint
+param       dstY1       GLint
+param       mask        GLbitfield
+param       filter      GLenum
+paramprop   filter      GL_LINEAR GL_NEAREST
+category    GL_EXT_framebuffer_blit
+chromium    extpack
+
+# GL_EXT_framebuffer_blit
+name        BlitFramebuffer
+alias       BlitFramebufferEXT
+return      void
+param       srcX0       GLint
+param       srcY0       GLint
 param       srcX1       GLint
 param       srcY1       GLint
 param       dstX0       GLint
--- virtualbox-5.2.8-dfsg.orig/src/VBox/GuestHost/OpenGL/include/chromium.h
+++ virtualbox-5.2.8-dfsg/src/VBox/GuestHost/OpenGL/include/chromium.h
@@ -61,6 +61,13 @@
 #endif
 
 #include <GL/gl.h>
+/* Quick fix so as not to update the version of glext.h we provide. */
+#ifdef GL_GLEXT_PROTOTYPES
+GLAPI void APIENTRY glFramebufferTexture2D (GLenum, GLenum, GLenum, GLuint, GLint);
+GLAPI void APIENTRY glBindFramebuffer (GLenum, GLuint);
+GLAPI void APIENTRY glBlitFramebuffer (GLint, GLint, GLint, GLint, GLint, GLint, GLint, GLint, GLbitfield, GLenum);
+GLAPI void APIENTRY glGenFramebuffers (GLsizei, GLuint *);
+#endif
 
 #ifndef WINDOWS
 # ifndef RT_OS_WINDOWS /* If we don't need it in ring-3, we probably not need it in ring-0 either (triggers warnings). */
